### PR TITLE
docs(solid-testing-library): add solid testing library docs

### LINKED ISSUE #1364: Add solid-testing-library to docs
**Is your feature request related to a problem? Please describe.**

The testing library docs don’t currently include solid-testing-library.

**Describe the solution you'd like** 

The testing library docs should include solid-testing-library, perhaps in the frameworks or ecosystem sections, with at minimum an introduction page that explains who to install and basic usage.

**Describe alternatives you've considered** 

N/A

**Additional context**

See this exchange on X (formally Twitter): https://x.com/kentcdodds/status/1753651787259863103?s=46&t=GPHIl0qoqoGb9U1BpHeF7g

GitHub: https://github.com/solidjs/solid-testing-library

NPM: https://www.npmjs.com/package/@solidjs/testing-library

### PR DESCRIPTION
**Pull request by danieljcafonso**
Added a Solid Testing Library section!

Note: this is mostly a port of https://github.com/solidjs/solid-testing-library current docs. Please let me know if changes are needed :)

Closes #1364

### CODE REVIEW

File: docs/solid-testing-library/intro.mdx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+2.  Specific to a testing framework.
+
+If you using Jest we recommend using
+[solid-jest](https://github.com/solidjs/solid-jest) to properly resolve the
+browser version of Solid as Jest will default to the server version when run in
+Node.
+
+💡 If you are using Jest or vitest, you may also be interested in installing
+`@testing-library/jest-dom` so you can use
+[the custom jest matchers](https://github.com/testing-library/jest-dom).
```
Comment (timdeschryver) [inline docs/solid-testing-library/intro.mdx:52]: Instead of linking to GitHub, this can also redirect to `docs/ecosystem-jest-dom`.

File: sidebars.js [hunk 2]
```diff
@@ -165,6 +165,12 @@ module.exports = {
             'react-native-testing-library/setup',
           ],
         },
+        {
```
Comment (timdeschryver) [inline sidebars.js:168]: Could you also add a redirect to this page at https://github.com/testing-library/testing-library-docs/blob/main/netlify.toml#L85 please?

File: docs/solid-testing-library/api.mdx [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+---
+
+Due to being inspired by the [preact-testing-library][preact-docs] you can check
+its page for more information.
+
+[preact-docs]: ../preact-testing-library/intro.mdx
+
+There are several key differences, to be aware of.
+
+## Differences from the Preact Testing Library
```
Comment (timdeschryver) [inline docs/solid-testing-library/api.mdx:14]: I don't know if this is needed.
Most Testing Libraries are all inspired by DOM testing library/each other.

Thoughts?

File: docs/solid-testing-library/api.mdx [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+otherwise only get `undefined` and wonder why this is the case.
+
+Solid.js supports
+[custom directives](https://www.solidjs.com/docs/latest/api#use___), which is a
+convenient pattern to tie custom behavior to elements, so we also have a
+`renderDirective` call, which augments `renderHook` to take a directive as first
+argument, accept an `initialValue` for the argument and a `targetElement`
+(string, HTMLElement or function returning a HTMLElement) in the `options` and
+also returns `arg` and `setArg` to read and manipulate the argument of the
+directive.
```
Comment (timdeschryver) [inline docs/solid-testing-library/api.mdx:104]:
```suggestion
Solid.js supports
[custom directives](https://www.solidjs.com/docs/latest/api#use___), which is a
convenient pattern to tie custom behavior to elements, so we also have a
`renderDirective` call, which augments `renderHook` to take a directive as first
argument, accept an `initialValue` for the argument and a `targetElement`
(string, HTMLElement or function returning an HTMLElement) in the `options` and
also returns `arg` and `setArg` to read and manipulate the argument of the
directive.
```

**Review by timdeschryver — COMMENTED** [review]
Looks good I did leave a few remarks/questions but these are more suggestions.

File: docs/solid-testing-library/api.mdx (hunk 3 above)
Comment (MatanBobi) [inline docs/solid-testing-library/api.mdx:14]: We usually also keep a table of contents for all the API's in this file with links (these can be auto generated AFAIR) :)

Comment (MatanBobi) [reply docs/solid-testing-library/api.mdx:14]: I think that this doc is mainly showing comparisons (as can be seen in the `render` section, I'm still not sure it's needed though, I believe we can remove the "preact" examples and just keep the "solid" examples here.. Wdyt?

File: docs/solid-testing-library/api.mdx [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+// With solid-testing-library
+const results = render(() => <YourComponent />, options)
+```
+
+Solid.js does _not_ re-render, it merely executes side effects triggered by
+reactive state that change the DOM, therefore there is no `rerender` method. You
+can use global signals to manipulate your test component in a way that causes it
+to update.
+
+In extension of the original API, the render function of this testing library
```
Comment (MatanBobi) [inline docs/solid-testing-library/api.mdx:36]: This is really a nit:

```suggestion
In addition to the original API, the render function of this testing library
```

File: docs/solid-testing-library/api.mdx [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+supports a convenient `location` option that will set up an in-memory router
+pointing at the specified location. Since this setup is not instantaneous, you
```
Comment (MatanBobi) [inline docs/solid-testing-library/api.mdx:38]: Is this the correct meaning? :)
```suggestion
pointing at the specified location. Since this setup is not synchronous, you
```

File: docs/solid-testing-library/api.mdx [hunk 7]
```diff
@@ -0,0 +1,180 @@
...
+      <Route path="/" component={() => <p>Start</p>} />
+    </Routes>
+  )
+  const {findByText} = render(() => <App />, {location: 'ids/1234'})
+  expect(await findByText('Id: 1234')).not.toBeFalsy()
+})
+```
+
+It uses `@solidjs/router`, so if you want to use a different router, you should
+consider the `wrapper` option instead. If you attempt to use this without having
```
Comment (MatanBobi) [inline docs/solid-testing-library/api.mdx:55]: I think that this information should also be added to an FAQ but this is just an opinion and not a blocking comment :)

File: docs/solid-testing-library/api.mdx [hunk 8]
```diff
@@ -0,0 +1,180 @@
...
+the package installed, you will receive an error message.
+
+### `renderHook`
+
+Solid.js external reactive state does not require any DOM elements to run in, so
+our `renderHook` call to test hooks in the context of a component (if your hook
+does not require the context of a component, `createRoot` should suffice to test
+the reactive behavior; for convenience, we also have `createEffect`, which is
```
Comment (MatanBobi) [inline docs/solid-testing-library/api.mdx:63]: Maybe we should link to `createEffect` here? and then we'll be able to remove the "which is described later" sentence.

File: docs/solid-testing-library/api.mdx [hunk 9]
```diff
@@ -0,0 +1,180 @@
...
+If you are using a `wrapper` with `renderHook`, make sure it will **always**
+return `props.children` - especially if you are using a context with
+asynchronous code together with `<Show>`, because this is required to get the
+value from the hook and it is only obtained synchronously once and you will
+otherwise only get `undefined` and wonder why this is the case.
+
+Solid.js supports
+[custom directives](https://www.solidjs.com/docs/latest/api#use___), which is a
+convenient pattern to tie custom behavior to elements, so we also have a
+`renderDirective` call, which augments `renderHook` to take a directive as first
```
Comment (MatanBobi) [inline docs/solid-testing-library/api.mdx:100]: Should this also have an `h3`?

**Review by MatanBobi — COMMENTED** [review]
That's amazing, thank you so much @danieljcafonso!
I've left a few minor comments.

File: docs/solid-testing-library/intro.mdx (hunk 1 above)
Comment (danieljcafonso) [reply docs/solid-testing-library/intro.mdx:52]: will do! Thank you!

File: docs/solid-testing-library/api.mdx (hunk 3 above)
Comment (danieljcafonso) [reply docs/solid-testing-library/api.mdx:14]: adding a Toc

Comment (danieljcafonso) [reply docs/solid-testing-library/api.mdx:14]: Thank you :)

File: docs/solid-testing-library/api.mdx (hunk 8 above)
Comment (danieljcafonso) [reply docs/solid-testing-library/api.mdx:63]: will do :)

File: docs/solid-testing-library/api.mdx (hunk 7 above)
Comment (danieljcafonso) [reply docs/solid-testing-library/api.mdx:55]: I would say this information is relevant to keep on this part as it is something that impacts the usage of this render function. 🤔

File: sidebars.js (hunk 2 above)
Comment (danieljcafonso) [reply sidebars.js:168]: done :)

File: docs/solid-testing-library/api.mdx (hunk 3 above)
Comment (danieljcafonso) [reply docs/solid-testing-library/api.mdx]: done

**Review by timdeschryver — APPROVED** [review]
Thanks LGTM!
I'll leave this open for @MatanBobi

File: docs/solid-testing-library/api.mdx (hunk 7 above)
Comment (MatanBobi) [reply docs/solid-testing-library/api.mdx:53]: I agree on this one, that's why I said also, but we can iterate on this later if we'll find the need :)
Thanks!

**Review by MatanBobi — APPROVED** [review]
Thanks @danieljcafonso! Super excited for this one.
